packages/runtime/container-runtime/src/summarizerTypes.ts

```diff
@@ -220,11 +266,14 @@ export interface ISummarizer
     run(onBehalfOf: string, options?: Readonly<Partial<ISummarizerOptions>>): Promise<void>;
     updateOnBehalfOf(onBehalfOf: string): void;
 
-    /** Attempts to generate a summary on demand. */
-    summarizeOnDemand(
-        reason: string,
-        options: Omit<ISubmitSummaryOptions, "summaryLogger">,
-    ): OnDemandSummarizeResult;
+    /** Attempts to generate a summary on demand. If already running, takes no action. */
+    summarizeOnDemand(options: IOnDemandSummarizeOptions): OnDemandSummarizeResult;
```

Comment:
Can you please add an @returns here? Basically saying that if a summarize is in progress, it returns a promise that is resolved when the summarize completes and another attempt should be made after its done.

Similar comments for `enqueueSummarize`

Reply:
Added.. they are pretty big comments though